test/list_notifier_test.dart

```diff
@@ -0,0 +1,157 @@
+import 'package:flutter_test/flutter_test.dart';
+
+import '../lib/listenable_collections.dart';
+
+void main() {
+  ListNotifier list;
+
+  setUp(() {
+    list = ListNotifier(data: [1, 2, 3]);
+  });
+
+
+  test("Elements get swapped correctly", () {
+    list.swap(0, 2);
+
+    expect(list.value, [3, 2, 1]);
+  });
```

Comment:
Could you add a check that the notifyListener is called here too?

Reply:
Sure, will do it later today